### PR TITLE
Correct packages and fpm as checkinstall alternative for RedHat and CentOS

### PR DESCRIPTION
**Pull request by JulioECastro**
I send you a proposal to solve some problems with RedHat and CentOS.
I used gem FPM as alternative to Checkinstall.

Regards.-

### CODE REVIEW

**Comment by gravyboat** [comment]
Hmm, I think we could improve on this if the package stuff was moved into the map and then loaded from there. Building from scratch should also be in a different state I think if we can manage it.

**Comment by myoung34** [comment]
@gravyboat  Can we get this merged upstream/find out what's blocking it?

File: node/source.sls [hunk 1]
```diff
@@ -48,11 +54,42 @@ make-node:
...
+    - sources:
+      - node: /usr/src/node-v{{ version }}/node-{{ version }}-1.x86_64.rpm
+    - onchanges:
+      - cmd: fpm-node
+{% else %}
+
 install-node:
   cmd.run:
     - cwd: /usr/src/node-v{{ version }}
     - name: checkinstall --install=yes --pkgname=node --pkgversion "{{ version }}" --default
```
Comment (myoung34) [inline node/source.sls]: @JulioECastro this needs to be pulled out for centos, `checkinstall` is not installed

**Comment by myoung34** [comment]
@JulioECastro you should use `grains['os_family']` instead of `grains['os']` to support derivatives like AWS Linux

```[ec2-user@ip-10-0-3-149 ~]$ sudo salt-call grains.get os
local:
    Amazon
[ec2-user@ip-10-0-3-149 ~]$ sudo salt-call grains.get os_family
local:
    RedHat```

**Comment by myoung34** [comment]
Tested it on AWS Linux (centos 6 fork) and it works as expected.

**Pull request merged**